show/main.py

```diff
@@ -160,10 +175,7 @@ def cli(ctx):
 cli.add_command(warm_restart.warm_restart)
 
 # Add greabox commands only if GEARBOX is configured
-# TODO: Find a cleaner way to do this
-app_db = SonicV2Connector(host='127.0.0.1')
-app_db.connect(app_db.APPL_DB)
-if app_db.keys(app_db.APPL_DB, '_GEARBOX_TABLE:phy:*'):
+if is_gearbox_configured():
```

Comment:
is_gearbox_configured API will be False in a couple of cases:
1. if gearsyncd still initialize the gearbox;
2. if something was wrong with configuration files while setting up gearbox;
3. if something wrong with gearsyncd or some part of gearbox daemons stack.

Unfortunately, I have not any information about how the CLI commands could be designed. Maybe, it would be better to register gearbox commands always, but return something like this: "Gearbox is unconfigured" untill the gearbox will be configured.

@gechiang, maybe, you know somebody who could have some information about this?

Reply:
@sydlogan Can you help comment on this?  It looks like you were the original author for this change.
My understanding is that for Platform that does not have GEARBOX enabled the default is always set _GEARBOX_TABLE:GearboxConfigDone to success True.  Which means the check that @maksymbelei95 is trying to use will always end up adding the CLI even if GEARBOX is not there.
We need to have another way to detect that GEARBOX is not installed so we can skip hitting this error that gets logged onto syslog each time someone issue show interface status or show interface counters...

Reply:
@maksymbelei95 in the interest of time can you revert back to the original way you were checking this but slightly different where your search is to include the PHY keywords?
keys = app_db.keys(app_db.APPL_DB, '_GEARBOX_TABLE:phy:*')
If this keys dictionary is not empty then GEARBOX is configured and you can register the gearbox show cmd 
this way it should avoid hitting the warning and maintain the same functionality as before.

Reply:
@gechiang, I have checked your proposition locally. I have changed is_gearbox_configured() like this:
```
def is_gearbox_configured():
    """
    Checks whether Gearbox is configured or not
    """
    app_db = SonicV2Connector()
    app_db.connect(app_db.APPL_DB)

    keys = app_db.keys(app_db.APPL_DB, '*')

    # If any _GEARBOX_TABLE:phy:* records present in APPL_DB, then the gearbox is configured
    if any(re.match(GEARBOX_TABLE_PHY_PATTERN, key) for key in keys):
        return True
    else:
        return False
```
,where `GEARBOX_TABLE_PHY_PATTERN = r"_GEARBOX_TABLE:phy:*"`.

The unit tests have passed, my SN2700 without gearbox does not log any warnings.
Previously, you have rejected the approach with ".keys" due to its complexity. If you confirm that the approach is ok, I will update the commit and ping you.

Reply:
@maksymbelei95 
Please go ahead.  With your proposed new change.
Thanks!

Reply:
@gechiang,
I have updated the change. Could you check it?